src/client_side.cc

```diff
@@ -3344,24 +3345,17 @@ clientHttpConnectionsOpen(void)
                                (s->workerQueues ? COMM_REUSEPORT : 0);
 
         typedef CommCbFunPtrCallT<CommAcceptCbPtrFun> AcceptCall;
-        if (s->transport.protocol == AnyP::PROTO_HTTP) {
-            // setup the subscriptions such that new connections accepted by listenConn are handled by HTTP
-            RefCount<AcceptCall> subCall = commCbCall(5, 5, "httpAccept", CommAcceptCbPtrFun(httpAccept, CommAcceptCbParams(NULL)));
-            Subscription::Pointer sub = new CallSubscription<AcceptCall>(subCall);
-
-            AsyncCall::Pointer listenCall = asyncCall(33,2, "clientListenerConnectionOpened",
-                                            ListeningStartedDialer(&clientListenerConnectionOpened, s, Ipc::fdnHttpSocket, sub));
-            Ipc::StartListening(SOCK_STREAM, IPPROTO_TCP, s->listenConn, Ipc::fdnHttpSocket, listenCall);
-
-        } else if (s->transport.protocol == AnyP::PROTO_HTTPS) {
-            // setup the subscriptions such that new connections accepted by listenConn are handled by HTTPS
-            RefCount<AcceptCall> subCall = commCbCall(5, 5, "httpsAccept", CommAcceptCbPtrFun(httpsAccept, CommAcceptCbParams(NULL)));
-            Subscription::Pointer sub = new CallSubscription<AcceptCall>(subCall);
-
-            AsyncCall::Pointer listenCall = asyncCall(33, 2, "clientListenerConnectionOpened",
-                                            ListeningStartedDialer(&clientListenerConnectionOpened,
-                                                    s, Ipc::fdnHttpsSocket, sub));
-            Ipc::StartListening(SOCK_STREAM, IPPROTO_TCP, s->listenConn, Ipc::fdnHttpsSocket, listenCall);
+        const auto &protocol = s->transport.protocol;
+        if (protocol == AnyP::PROTO_HTTP || protocol == AnyP::PROTO_HTTPS) {
+            const auto isHttp = protocol == AnyP::PROTO_HTTP;
+            CallBack(s, [&] {
```

Comment:
CallBack() does the right thing here, but the name is misleading. We need to add a more general RunInContext() or similar function that does the same thing. I would be happy to add that function and adjust CallBack() docs/implementation accordingly, but I think that is best done in a separate PR. If you agree, then I suggest documenting the problem here and then describing this solution plan in a GitHub comment. If Amos insists on those changes done at the same time/PR or in a prerequisite PR, we would do that.

```suggestion
            // TODO: Define RunInContext() to provide CallBack()-like functionality
            // for cases not dealing with calling a callback, like this one.
            CallBack(s, [&] {
```